### PR TITLE
Second argument is replaced upon completing the first argument

### LINKED ISSUE #462: Second argument is replaced upon completing the first argument in a method call
**Moved from https://github.com/Microsoft/vscode-java-pack/issues/12:**

- VSCode Version: Latest Insiders
- OS Version: Mac OS

Steps to Reproduce:
1. Insert cursor before the first argument (that accesses some property) to the method.
2. Start typing new argument that also accesses property. 
3. On suggestion appearance, press Tab to enter suggestion value
It replaces the old first argument instance with the property from the new argument. 

vscode

Sample code:
```java
public class Test {
    public int testInt;

    public boolean method(int x, int y, int z) {
        return true;
    }
}

public class Main {
    public static void main(String[] args) {
        Test t = new Test();
        t.method(t.testInt, t.testInt);
    }
}
```

### PR DESCRIPTION
**Pull request by snjeza**
Fixes #462 
Requires https://github.com/eclipse/eclipse.jdt.ls/pull/619

Signed-off-by: Snjezana Peco <snjezana.peco@redhat.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -182,6 +182,12 @@
           "description": "Enable/disable the 'auto build'",
           "scope": "window"
         },
+        "java.completion.insert": {
+          "type": "boolean",
+          "default": true,
+          "description": "Completion inserts/overwrites",
```
Comment (gorkem) [inline package.json]: I think we need to explain on description what does true mean?

**Comment by snjeza** [comment]
@gorkem I have updated the PR.

**Comment by fbricon** [comment]
Honestly, from a user standpoint, I think it'd be clearer to have a `java.completion.overwrite` pref, that said 

> When set to `true`, code completion overwrites the current text. When set to `false`, code is simply  added instead.

**Comment by snjeza** [comment]
@gorkem @fbricon I have updated the PR.

**Comment by fbricon** [comment]
completion keeps overwriting the text, regardless of the preference

File: package.json [hunk 2]
```diff
@@ -182,6 +182,12 @@
           "description": "Enable/disable the 'auto build'",
           "scope": "window"
         },
+        "java.completion.overwrite ": {
```
Comment (fbricon) [inline package.json]: beware of the trailing space, it messes up configuration

**Comment by snjeza** [comment]
@gorkem @fbricon I have updated the PR.

**Pull request merged**